csrc/scheduler/greedy.cpp

```diff
@@ -804,12 +900,20 @@ void propagateReshape(Fusion* fusion) {
 void insertCopies(Fusion* fusion, GreedyParams& greedy_params) {
   for (auto expr : getAllConstrainedOps(fusion)) {
     if (expr->isA<ScatterOp>()) {
+      auto inp_tv = ir_utils::getTvInput(expr);
       auto out_tv = ir_utils::getTvOutput(expr);
       if (!out_tv->uses().empty()) {
-        out_tv->cacheAfter(
+        auto cache = out_tv->cacheAfter(
             LoadStoreOpType::Set,
             CacheOp::Unspecified,
             /*propagate_allocation_domain=*/false);
+        // Since the cache, which is a copy of the scatter output,
+        // needs to be scheduled as a constrained tensor, add its
+        // consumer parameter. The same parater as the producer
+        // parameter of the index tensor should be used.
```

Comment:
for scatter, I think the producer parameter is the buffer parameter. while the comment above is suggesting that we are using the parameter of the index tensor.